packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -164,51 +164,44 @@ class PointerEventResampler {
 
   void _dequeueAndSampleNonHoverOrMovePointerEventsUntil(
       Duration sampleTime,
+      Duration nextSampleTime,
       HandleEventCallback callback,
   ) {
-    while (_queuedEvents.isNotEmpty) {
-      final PointerEvent event = _queuedEvents.first;
+    Duration endTime = sampleTime;
+    // Scan queued events to determine end time.
+    final Iterator<PointerEvent> it = _queuedEvents.iterator;
+    while (it.moveNext()) {
+      final PointerEvent event = it.current;
 
       // Potentially stop dispatching events if more recent than `sampleTime`.
       if (event.timeStamp > sampleTime) {
-        // Stop if event is not up or removed. Otherwise, continue to
-        // allow early processing of up and remove events as this improves
-        // resampling of these events, which is important for fling
-        // animations.
-        if (event is! PointerUpEvent && event is! PointerRemovedEvent) {
+        // Definitely stop if more recent than `nextSampleTime`.
+        if (event.timeStamp > nextSampleTime) {
```

Comment:
Nit: `event.timeStamp >= nextSampleTime`? It probably doesn't matter much in practice, but logically I think you don't have to resample the next up event if it's right on the next sample time.

Reply:
Done.